mapviz_plugins/src/move_base_plugin.cpp

```diff
@@ -0,0 +1,338 @@
+            break;
+
+        case actionlib::SimpleClientGoalState::PREEMPTED:
+            PrintWarningHelper( ui_.status, state.toString() );
+            monitoring_action_state_ = false;
+            break;
+
+        case actionlib::SimpleClientGoalState::ACTIVE:
+            PrintInfoHelper( ui_.status, state.toString() );
+            break;
+
+        case actionlib::SimpleClientGoalState::SUCCEEDED:
+            PrintInfoHelper( ui_.status, state.toString() );
+            monitoring_action_state_ = false;
+            break;
+
+        case actionlib::SimpleClientGoalState::REJECTED:
+        case actionlib::SimpleClientGoalState::ABORTED:
+            case actionlib::SimpleClientGoalState::LOST:
+            PrintErrorHelper( ui_.status, state.toString() );
+            monitoring_action_state_ = false;
+            break;
+        }
+    }
+}
+
+
+bool MoveBasePlugin::handleMousePress(QMouseEvent* event)
+{
+    bool init_checked = ui_.pushButtonInitialPose->isChecked();
+    bool goal_checked = ui_.pushButtonGoalPose->isChecked();
+    if( !init_checked && !goal_checked)
+    {
+        return false;
+    }
+
+    if (event->button() == Qt::LeftButton)
+    {
+        is_mouse_down_ = true;
+        arrow_angle_ = 0;
+        arrow_tail_position_= map_canvas_->MapGlCoordToFixedFrame( event->localPos() );
+        return true;
+    }
+    return false;
+}
+
+bool MoveBasePlugin::handleMouseMove(QMouseEvent* event)
+{
+    if (is_mouse_down_)
+    {
+        QPointF head_pos = map_canvas_->MapGlCoordToFixedFrame( event->localPos() );
+        arrow_angle_ = atan2( head_pos.y() - arrow_tail_position_.y(),
+                              head_pos.x() - arrow_tail_position_.x() );
+    }
+    return false;
+}
+
+bool MoveBasePlugin::handleMouseRelease(QMouseEvent* event)
+{
+    if( !is_mouse_down_ )
+    {
+        return false;
+    }
+
+    is_mouse_down_ = false;
+
+    bool init_checked = ui_.pushButtonInitialPose->isChecked();
+    bool goal_checked = ui_.pushButtonGoalPose->isChecked();
+    if( !init_checked && !goal_checked)
+    {
+        return false;
+    }
+
+    tf::Quaternion quat = tf::createQuaternionFromYaw(arrow_angle_);
+
+    if( goal_checked ){
+
+        move_base_msg_.action_goal.header.frame_id = target_frame_;
+        move_base_msg_.action_goal.header.stamp = ros::Time::now();
+        move_base_msg_.action_goal.goal_id.stamp = move_base_msg_.action_goal.header.stamp;
+        move_base_msg_.action_goal.goal_id.id = "mapviz_goal";
+        move_base_msg_.action_goal.goal.target_pose.header = move_base_msg_.action_goal.header;
+
+        geometry_msgs::Pose& pose = move_base_msg_.action_goal.goal.target_pose.pose;
+        pose.position.x = arrow_tail_position_.x();
+        pose.position.y = arrow_tail_position_.y();
+        pose.position.z = 0.0;
+        tf::quaternionTFToMsg( quat, pose.orientation );
+
+        move_base_client_.sendGoal(move_base_msg_.action_goal.goal);
```

Comment:
I see that you're calling a `move_base` service here, but it looks like the rviz "2D Nav Goal" button publishes a `geometry_msgs/PoseStamped` message.  Why the difference?

Reply:
the RViz interface is "send and forget". It means that I can't get any feedback about the current execution (getStatus) and I won't be able to cancel the action.